Destination/Destination.php

```diff
@@ -3,15 +3,25 @@
 namespace Zenstruck\BackupBundle\Destination;
 
 use Psr\Log\LoggerInterface;
+use \Countable;
+use \IteratorAggregate;
 
 /**
  * @author Kevin Bond <kevinbond@gmail.com>
  */
-interface Destination
+interface Destination extends Countable, IteratorAggregate
```

Comment:
Ok, I looked at the output of `s3cmd ls <bucket>/prefix/` and it isn't as bad as I thought. Should be able to parse it easy enough with a bit of regex.

So, let's go with having the Destination Countable/IteratorAggregate.

I still would like `remove`/`list` methods.